keylime/da/attest.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/python3
+import argparse
+import sys
+from datetime import datetime
+
+from keylime import cloud_verifier_common, config, keylime_logging
+from keylime.da import record
+
+logger = keylime_logging.init_logging("durable_attestation_fetch_and_replay")
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-u",
+        "--uuid",
+        action="store",
+        dest="agent_uuid",
+        default="all",
+        help='UUID for the agent to attest (default "all")',
+    )
+    parser.add_argument(
+        "-s",
+        "--start",
+        action="store",
+        dest="start_date",
+        default="first",
+        help='start date for attestation window in either IS0 8601 format (e.g., "2008-09-03T20:56:35"), or keyworkd "first"/"none"',
+    )
+    parser.add_argument(
+        "-e",
+        "--end",
+        action="store",
+        dest="end_date",
+        default="last",
+        help='end date for attestation window in IS0 8601 format (e.g., "2008-09-03T21:56:35"), or keyworkd "last"',
+    )
+
+    args = parser.parse_args()
+
+    rmc = record.get_record_mgt_class(config.get("registrar", "durable_attestation_import", fallback=""))
+    if rmc:
+        rmc = rmc("registrar")
+
+        if args.end_date == "last":
+            end_date = rmc.end_of_times
+        else:
+            end_date = int(datetime.fromisoformat(args.end_date).strftime("%s"))
+
+        if args.start_date == "first":
+            start_date = rmc.start_of_times
+        elif args.start_date == "none":
+            start_date = end_date - 1
+        else:
+            start_date = int(datetime.fromisoformat(args.start_date).strftime("%s"))
+
+        if end_date != rmc.end_of_times or start_date != rmc.end_of_times - 1:
+            logger.info(
+                "=> Selected date range for attestation records is: start %d (%s) and end %d (%s)",
+                start_date,
+                datetime.fromtimestamp(start_date).strftime("%Y-%m-%dT%H:%M:%S.%f%z"),
+                end_date,
+                datetime.fromtimestamp(end_date).strftime("%Y-%m-%dT%H:%M:%S.%f%z"),
+            )
+        else:
+            logger.info("=> Will only extract the very last attestation record found on the persistent store")
+
+        if args.agent_uuid.lower() == "all":
+            logger.info("=> Getting a list of agents registered on the persistent store\n")
+            _agent_list = rmc.agent_list_retrieval("auto")
```

Comment:
We should drop the `_` prefix in all the local variables in the new code. As commented in a different PR, in Python the `_` prefix has a meaning for fields or names in modules. Because of that a Python developer can see this as confusing.

More info: https://peps.python.org/pep-0008/#descriptive-naming-styles